studio/LonaStudio/ForeignTypes.swift

```diff
@@ -17,6 +17,7 @@ public typealias TextStyleList = [CSTextStyle]?
 public typealias TextStyleHandler = ((CSTextStyle?) -> Void)?
 
 public typealias StringHandler = ((String) -> Void)?
+public typealias NumberHandler = ((CGFloat) -> Void)?
```

Comment:
So I'm wondering if we could generate those types automatically: it's quite painful to define the function every time, and prone to error. Moreover, the information is already in `types.json`:
```json
  {
      "name": "Named",
      "alias": "NumberHandler",
      "of": {
        "name": "Function",
        "parameters": [
          {
            "label": "_",
            "type": "Number"
          }
        ]
      }
    }
```

That's why I chose to create a new alias, because I thought we could generate them automatically but I agree that `ForeignTypes.swift` is a temp solution

Reply:
I agree that it's painful and error-prone to define the function each time... but I think it's still the best solution at the moment. If we define the function each time, we don't need to add anything to `types.json`, since there's no need to use the `NumberHandler` typealias in Swift -- we could take this out if we're not using it anywhere. 

Generating types from types.json is an option, but I feel like we shouldn't have to define custom types for every function. It seems like it'd get out of hand and be hard to remember what to use if we had to define a separate alias for every combination of parameter/return types, `(Int)`, `(Int, Int)`, `(Int, CGFloat)`, `(CGFloat, Int)`, `(Int, Int, Int)` and so on.

I think a better solution might be to make it extremely easy to specify custom function types. If I think of writing code, I almost always prefer writing out `(String) -> Void` versus defining a typealias. I think that's because it's so quick to type out or copy & paste, and there's no mental overhead of remembering what another typealias means. We could probably make it similarly easy by adding custom function types to the type dropdown that have been used elsewhere in the project, plus maybe a handful of common ones even if they aren't used in the project. We could also make types copy & pastable. I'm leaning in that direction. If the UI for the type picker/definer were nicer/easier to use, that would help a lot too.

Reply:
Let's just leave this as-is though for now, since you're right that `NumberHandler` would be a hassle to define in like 20 places.

Reply:
> I almost always prefer writing out (String) -> Void versus defining a typealias

That's interesting. I'd say that's true when the function is "standalone" but here it's in the case of passing a function to a sub-component. It's that sub-component which really care about the type of the function, the component itself doesn't care at all.

Reply:
Maybe we should have an option to generate the parameter and the logic to pass it down to a sub-component in one click

Reply:
I like that direction a lot. Wiring up parameters and logic are way too much work. I feel like a bunch of that could be mostly automated.